python/librmm/pyproject.toml

```
@@ -55,7 +55,8 @@ requires = [
 [tool.scikit-build]
 build-dir = "build/{wheel_tag}"
 cmake.build-type = "Release"
-cmake.minimum-version = "3.26.4"
+cmake.version = "CMakeLists.txt"
+minimum-version = "0.10"
```

As pointed out by Henry ( https://github.com/rapidsai/build-planning/issues/58#issuecomment-2274214167 ), we can configure `scikit-build-core` to retrieve the minimum version from `build-system.requires`

This is also mentioned in:

* [This blogpost]( https://iscinumpy.dev/post/scikit-build-core-0-10/ )
* [The scikit-build-core docs]( https://scikit-build-core.readthedocs.io/en/latest/configuration.html#minimum-version-defaults )

Here is what this change would entail:

```suggestion
minimum-version = "build-system.requires"
```

We probably want to roll this out across the existing PRs

cc @jameslamb (for awareness)

---

Agreed! I intentionally hadn't included this in these PRs last night, when we were confronting the possibility of them being last-minute changes for the 24.08 release. Wanted to change as little as was necessary to get things working again.

Now that that's not the case, I totally support this. Really nice feature 🤩 .

I'll go apply this across the other PRs if it works here in `rmm`.

---

Looks like this worked well! I've updated all the other PRs from https://github.com/rapidsai/build-planning/issues/58

---

Thanks James! 🙏